boa_engine/src/builtins/intl/mod.rs

```diff
@@ -144,3 +147,566 @@ impl Intl {
+
+        // i. If options has a field [[<key>]], then
+        if options.properties.contains_key(key) {
+            // i. Let optionsValue be options.[[<key>]].
+            // TODO ii. Assert: Type(optionsValue) is either String, Undefined, or Null.
+            let mut options_value = options
+                .properties
+                .get(key)
+                .unwrap_or(&JsValue::undefined())
+                .clone();
+
+            // iii. If Type(optionsValue) is String, then
+            if options_value.is_string() {
+                // TODO 1. Let optionsValue be the string optionsValue after performing the
+                // algorithm steps to transform Unicode extension values to canonical syntax
+                // per Unicode Technical Standard #35 LDML § 3.2.1 Canonical Unicode Locale
+                // Identifiers, treating key as ukey and optionsValue as uvalue productions.
+
+                // TODO 2. Let optionsValue be the string optionsValue after performing the
+                // algorithm steps to replace Unicode extension values with their canonical
+                // form per Unicode Technical Standard #35 LDML § 3.2.1 Canonical Unicode
+                // Locale Identifiers, treating key as ukey and optionsValue as uvalue
+                // productions.
+
+                // 3. If optionsValue is the empty String, then
+                let options_val_str = options_value
+                    .to_string(context)
+                    .unwrap_or_else(|_| JsString::new(""))
+                    .to_string();
+                if options_val_str.is_empty() {
+                    // a. Let optionsValue be "true".
+                    options_value = JsValue::String(JsString::new("true"));
+                }
+            }
+
+            // iv. If keyLocaleData contains optionsValue, then
+            let options_val_str = options_value
+                .to_string(context)
+                .unwrap_or_else(|_| JsString::new(""))
+                .to_string();
+            if js_arr_contains_str(&key_locale_data, &options_val_str, context) {
+                // 1. If SameValue(optionsValue, value) is false, then
+                if !options_value.eq(&value) {
+                    // a. Let value be optionsValue.
+                    value = options_value;
+
+                    // b. Let supportedExtensionAddition be "".
+                    supported_extension_addition = JsString::new("");
+                }
+            }
+        }
+
+        // j. Set result.[[<key>]] to value.
+        result.properties.insert(key.clone(), value);
+
+        // k. Append supportedExtensionAddition to supportedExtension.
+        supported_extension = JsString::concat(supported_extension, &supported_extension_addition);
+    }
+
+    // 10. If the number of elements in supportedExtension is greater than 2, then
+    let ext_elements = unicode_extension_components(&supported_extension);
+    if ext_elements.keywords.len() > 2 {
```

Comment:
This seems to differ from the spec no? Should this not just be `supported_extension.len() > 2`

Reply:
I misinterpreted the specification. Fixed https://github.com/boa-dev/boa/pull/2036/files#diff-1dd4af22d3a6142a8060d056e9d0489d84d8a4d9c68a8f6e7b359b8d3ab35499R647